src/Features/Core/Portable/Workspace/BackgroundParser.cs

```diff
@@ -178,8 +178,12 @@ private void ParseDocumentAsync(Document document)
 
             var cancellationToken = cancellationTokenSource.Token;
 
+            // once task runs, we never cancel running parsing request. this will make sure 
+            // chained incremental parsing request will always get shorten by BG
+            //
+            // we are alraedy running in BG, you no need to run async version
             var task = _taskScheduler.ScheduleTask(
-                () => document.GetSyntaxTreeAsync(cancellationToken),
+                () => document.GetSyntaxTreeSynchronously(CancellationToken.None),
                 "BackgroundParser.ParseDocumentAsync",
                 cancellationToken);
```

Comment:
should this cancellation token be here?   I imagine it's ok.  Is that effectively saying: if i haven't even kicked off the task, i don't even need to bother with it?

Reply:
yep. the later one will be invoked since only reason it gets cancelled is either there is new request or file is removed/closed.